internal/integration/rhp/v4/rhp4_test.go

```diff
@@ -0,0 +1,1383 @@
+		t.Fatal(err)
+	}
+	revised := fundResult.Revision
+
+	renterOutputValue := revision.Revision.RenterOutput.Value.Sub(accountFundAmount)
+	hostOutputValue := revision.Revision.HostOutput.Value.Add(accountFundAmount)
+
+	// verify the contract was modified correctly
+	switch {
+	case fundResult.Balances[0].Account != account:
+		t.Fatalf("expected %v, got %v", account, fundResult.Balances[0].Account)
+	case fundResult.Balances[0].Balance != accountFundAmount:
+		t.Fatalf("expected %v, got %v", accountFundAmount, fundResult.Balances[0].Balance)
+	case !fundResult.Usage.RenterCost().Equals(accountFundAmount):
+		t.Fatalf("expected %v, got %v", accountFundAmount, fundResult.Usage.RenterCost())
+	case !revised.HostOutput.Value.Equals(hostOutputValue):
+		t.Fatalf("expected %v, got %v", hostOutputValue, revised.HostOutput.Value)
+	case !revised.RenterOutput.Value.Equals(renterOutputValue):
+		t.Fatalf("expected %v, got %v", renterOutputValue, revised.RenterOutput.Value)
+	case !revised.MissedHostValue.Equals(revision.Revision.MissedHostValue):
+		t.Fatalf("expected %v, got %v", revision.Revision.MissedHostValue, revised.MissedHostValue)
+	case revised.RevisionNumber != revision.Revision.RevisionNumber+1:
+		t.Fatalf("expected %v, got %v", revision.Revision.RevisionNumber+1, revised.RevisionNumber)
+	}
+
+	revisionSigHash := cs.ContractSigHash(revised)
+	if !renterKey.PublicKey().VerifyHash(revisionSigHash, revised.RenterSignature) {
+		t.Fatal("revision signature verification failed")
+	} else if !hostKey.PublicKey().VerifyHash(revisionSigHash, revised.HostSignature) {
+		t.Fatal("revision signature verification failed")
+	}
+
+	// verify the account balance
+	balance, err = rhp4.RPCAccountBalance(context.Background(), transport, account)
+	if err != nil {
+		t.Fatal(err)
+	} else if !balance.Equals(accountFundAmount) {
+		t.Fatalf("expected %v, got %v", accountFundAmount, balance)
+	}
+}
+
+func TestReadWriteSector(t *testing.T) {
+	n, genesis := testutil.V2Network()
+	hostKey, renterKey := types.GeneratePrivateKey(), types.GeneratePrivateKey()
+
+	hn := testutil.NewHostNode(t, hostKey, n, genesis, zap.NewNop())
+	cm := hn.Chain
+	w := hn.Wallet
+
+	results := make(chan error, 1)
+	if _, err := hn.Volumes.AddVolume(context.Background(), filepath.Join(t.TempDir(), "test.dat"), 10, results); err != nil {
+		t.Fatal(err)
+	} else if err := <-results; err != nil {
+		t.Fatal(err)
+	}
+
+	testutil.MineAndSync(t, hn, w.Address(), int(n.MaturityDelay+20))
+
+	transport := testRenterHostPair(t, hostKey, hn, zap.NewNop())
+
+	settings, err := rhp4.RPCSettings(context.Background(), transport)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fundAndSign := &fundAndSign{w, renterKey}
+	renterAllowance, hostCollateral := types.Siacoins(100), types.Siacoins(200)
+	formResult, err := rhp4.RPCFormContract(context.Background(), transport, cm, fundAndSign, cm.TipState(), settings.Prices, hostKey.PublicKey(), settings.WalletAddress, proto4.RPCFormContractParams{
```

Comment:
Might be interesting to have an additional test which doesn't append the data to a contract and just performs on temporary storage.

Reply:
Not sure I understand. This test doesn't append data to a contract?

Reply:
Oh guess I was just imagining it then 😅